model_specific/SM/weinberg_angle.cpp

```diff
@@ -103,20 +66,20 @@ Weinberg_angle::Weinberg_angle()
    , precision_goal(1.0e-8)
    , rho_hat(0.)
    , sin_theta(0.)
-   , mw_pole(0.)
-   , data()
-   , susy_contributions(true)
 {
 }
 
-void Weinberg_angle::enable_susy_contributions()
-{
-   susy_contributions = true;
-}
-
-void Weinberg_angle::disable_susy_contributions()
+Weinberg_angle::Weinberg_angle(
+   const standard_model::Standard_model* model_,
```

Comment:
I think we should throw an exception of `model_ == nullptr`.

Btw., why are you passing a pointer? Do you want to express that this parameter is somehow optional (if `model_ == nullptr`)? If not, then a reference would be better, I think.

Reply:
I need to think about this. I forgot how all of this was supposed to work. Naively, I'd say that it was simply supposed to be a copy of a constructor from `<model>_weinber_angle`, e.g.
```cpp
CLASSNAME::SM_weinberg_angle(
   const SM_mass_eigenstates* model_,
   const Sm_parameters& sm_parameters_)
   : model(model_)
   , sm_parameters(sm_parameters_)
{
}
```

Reply:
If there is no reason that the `model_` pointer can be `nullptr`, then I'd prefer to pass `model_` by const reference, not by pointer.